src/api/fuel.js

```diff
@@ -0,0 +1,531 @@
+  verifyKeyOwnership = async (challenge/*: string*/)/*: Promise<boolean>*/ =>
+    this.user.verifyKeyOwnership(challenge);
+  getAccountName = async ()/*: Promise<string>*/ => this.user.getAccountName();
+  getChainId = async ()/*: Promise<string>*/ => this.user.getChainId();
+  getKeys = async ()/*: Promise<string[]>*/ => this.user.getKeys();
+}
+
+// Auxiliar functions to validate with the user the use of the service
+/*
+interface Preference {
+  remember?: boolean;
+  approve?: boolean;
+}
+*/
+export default class GreymassFuelService {
+  static preferences/*: { [account: string]: Preference }*/ = {};
+  static globals/*: Record<string, (s: string) => string>*/ = null;
+  static save() {
+    localStorage.setItem(
+      'fuel_preferences',
+      JSON.stringify(GreymassFuelService.preferences)
+    );
+  }
+  static drop() {
+    localStorage.removeItem('fuel_preferences');
+  }  
+  static load() {
+    try {
+      GreymassFuelService.preferences = GreymassFuelService.preferences || {};
+      const str = localStorage.getItem('fuel_preferences');
+      if (str) {
+        GreymassFuelService.preferences = JSON.parse(str);
+      }
+    } catch (e) {
+      console.error('ERROR: ', e);
+    }
+  }
+
+  static setPreferences(account/*: string*/, p/*: Preference*/) {
+    GreymassFuelService.preferences[account] = {
+      ...GreymassFuelService.preferences[account],
+      ...p
+    };
+    if (GreymassFuelService.preferences[account].remember) {
+      GreymassFuelService.save();
+    } else {
+      GreymassFuelService.drop();
+    }
+  }
+  static setGlobals(g/*: Record<string, (s: string) => string>*/) {
+    GreymassFuelService.globals = g;
+  }
+}
+
+async function confirmWithUser(user/*: User*/, fees/*: string | null*/) {
+  const username = await user.getAccountName();
+  let mymodel/*: string[]*/ = [];
+  mymodel = [];
+
+
+  return new Promise((resolve, reject) => {
+    // Try and see if the user already answer (remembered)
+    if (
+      GreymassFuelService.preferences[username] &&
+      GreymassFuelService.preferences[username].remember
+    ) {
+      // ok, the user did. What's the answer?
+      if (GreymassFuelService.preferences[username].approve) {
+        resolve();
+      } else {
+        reject();
+      }
+      return;
+    }
+
+    const handler = function (approve/*: boolean*/) {
+      GreymassFuelService.setPreferences(username, { approve });
+      if (approve) resolve();
+      else reject();
+    };
+
+    // this are the normal texts for random wallet.
+    const cancel/*: string | boolean*/ = GreymassFuelService.globals.$t('api.reject');
+    const ok = GreymassFuelService.globals.$t('api.confirm');
+    let message = GreymassFuelService.globals.$t('api.greymass_fuel_message');
+
+    try {
+      if (typeof fees == 'string') {
+        message = GreymassFuelService.globals.$t('api.greymass_fuel_message_fees', { fees });
+      }
+    } catch (e) {}
```

Comment:
shouldn't we at least add a `console.error` here?

Reply:
I don't even know why the try-catch is there. I removed it.